hydropandas/io/water_connect.py

```diff
@@ -0,0 +1,388 @@
+    and grootheid_code.
+
+    Parameters
+    ----------
+    extent : list, tuple, numpy-array or None, optional
+        get water connect measurements within this extent
+        [xmin, xmax, ymin, ymax]
+    ObsClass : type
+        class of the observations, e.g. GroundwaterObs
+    tmin : str or None, optional
+        start time of observations. The default is None.
+    tmax : str or None, optional
+        end time of observations. The default is None.
+    only_metadata : bool, optional
+        if True download only metadata, significantly faster. The default
+        is False.
+    keep_all_obs : bool, optional
+        if False, only observations with measurements are kept. The default
+        is True.
+    location_gdf : GeoDataFrame, optional
+        geodataframe with the locations of the water drill holes you want to include.
+    **kwargs
+        additional keyword arguments are passed to the ObsClass.from_waterinfo()
+        method
+
+    Returns
+    -------
+    obs_list : list
+        list with Obs objects
+
+    """
+
+    if only_metadata and not keep_all_obs:
+        logger.error(
+            "you will get an empty ObsCollection with only_metadata is True and"
+            "keep_all_obs is False"
+        )
+
+    if location_gdf is None:
+        location_gdf = get_locations_gdf()
+
+    if extent is not None:
+        location_gdf = get_locations_within_extent(location_gdf, extent)
+
+    if location_gdf.empty:
+        msg = f"No water connect measurements found within extent {extent}"
+        logger.warning(msg)
+        return []
+
+    nr_obs_wells = len(location_gdf)
+    logger.info(f"downloading waterconnect data from {nr_obs_wells} observation points")
+
+    obs_list = []
+    for dh_no, meta_series in tqdm(
+        location_gdf.iterrows(), total=nr_obs_wells, desc="obs well"
+    ):
+        o = ObsClass.from_waterconnect(
+            dh_no,
+            meta_series,
+            tmin=tmin,
+            tmax=tmax,
+            only_metadata=only_metadata,
+            **kwargs,
+        )
+        if o.empty and not keep_all_obs:
+            continue
+
+        obs_list.append(o)
+
+    return obs_list
+
+
+def get_locations_gdf(keep_cols="all", update=False):
+    """get locations of water drillholes in Southern Australia from water connect.
+
+    Parameters
+    ----------
+    keep_cols : list or str, optional
+        if provided only these columns of the geodataframe are stored. If 'all' all the
+        columns are returned, by default 'all'.
+    update : bool, optional
+        if True new locations are downloaded and stored locally even if the locations
+        were already downloaded before. By default False
+
+    Returns
+    -------
+    geopandas.GeoDataFrame
+        locations of water drillholes in Southern Australia
+    """
+
+    # set directories
+    fdir = Path(user_data_dir("waterconnect", "hydropandas"))
```

Comment:
Wouldn't you want this to be a parameter?

Reply:
Yes! Fixed it

Reply:
Yes! Fixed it